components/api_server/src/initialization/database.py

```diff
@@ -50,3 +53,33 @@ def create_indexes(database: Database) -> None:
         [("metric_uuid", pymongo.ASCENDING), ("has_error", pymongo.ASCENDING), ("start", pymongo.DESCENDING)],
     )
     database.measurements.create_indexes([period_index, latest_measurement_index, latest_successful_measurement_index])
+
+
+def perform_migrations(database: Database) -> None:  # pragma: no cover-behave
+    """Perform database migrations."""
+    change_accessibility_violation_metrics_to_violations(database)
+
+
+def change_accessibility_violation_metrics_to_violations(database: Database) -> None:  # pragma: no cover-behave
+    """Replace accessibility metrics with the violations metric."""
+    # Added after Quality-time v5.5.0, see https://github.com/ICTU/quality-time/issues/562
+    for report in latest_reports(database):
+        changed = False
+        for subject in report["subjects"].values():
+            for metric in subject["metrics"].values():
+                if metric["type"] == "accessibility":
+                    change_accessibility_violations_metric_to_violations(metric)
+                    changed = True
+        if changed:
+            report_id = report["_id"]
+            del report["_id"]
+            database.reports.replace_one({"_id": report_id}, report)
+
+
+def change_accessibility_violations_metric_to_violations(metric: dict) -> None:  # pragma: no cover-behave
+    """Change the accessibility violations metric to violations metric."""
+    metric["type"] = "violations"
+    if not metric.get("name"):
```

Comment:
Why do we set name and unit when it is unset, shouldn't it default to a value from the data model?

Reply:
If the user has not changed the accessibility metric's name or unit they currently have the (default) name "Accessibility violations" and the (default) unit "accessibility violations". The migration code changes the name and unit of these metrics so as to minimise the change from the user's perspective. Without that, the accessibility metrics would be called "Violations" and have "violations" as unit after the migration.

Reply:
Ah, understandable. I thought it would make sense to have the name and unit automatically migrated :)